core/domain/exp_services_test.py

```diff
@@ -3057,3 +3057,148 @@ def test_reject_actioned_suggestion(self):
+            draft_change_list_exp_version=4).put()
+        # Exploration with no draft.
+        user_models.ExplorationUserDataModel(
+            id='%s.%s' % (self.USER_ID, self.EXP_ID3), user_id=self.USER_ID,
+            exploration_id=self.EXP_ID3).put()
+
+    def _empty_method(self, strict):
+        pass
+
+    def _get_exp_domain_object(self, unused_exp_id, unused_change_list):
+        return exp_domain.Exploration.create_default_exploration(
+            self.EXP_ID1, 'A title', 'A category')
+
+    def test_draft_cleared_after_change_list_applied(self):
+        exp_services.update_exploration(
+            self.USER_ID, self.EXP_ID1, self.draft_change_list, '')
+        exp_user_data = user_models.ExplorationUserDataModel.get_by_id(
+            '%s.%s' % (self.USER_ID, self.EXP_ID1))
+        self.assertIsNone(exp_user_data.draft_change_list)
+        self.assertIsNone(exp_user_data.draft_change_list_last_updated)
+        self.assertIsNone(exp_user_data.draft_change_list_exp_version)
+
+    def test_draft_version_valid_true(self):
+        self.assertTrue(exp_services.is_draft_version_valid(
+            self.EXP_ID1, self.USER_ID))
+
+    def test_draft_version_valid_false(self):
+        self.assertFalse(exp_services.is_draft_version_valid(
+            self.EXP_ID2, self.USER_ID))
+
+    def test_create_or_update_draft_older_draft_exists(self):
+        with self.swap(exp_domain.Exploration, 'validate',
+                       self._empty_method):
+            exp_services.create_or_update_draft(
+                self.EXP_ID1, self.USER_ID, self.NEW_CHANGELIST, 5,
+                self.NEWER_DATETIME)
+        exp_user_data = user_models.ExplorationUserDataModel.get(
+            self.USER_ID, self.EXP_ID1)
+        self.assertEqual(exp_user_data.exploration_id, self.EXP_ID1)
+        self.assertEqual(
+            exp_user_data.draft_change_list, self.NEW_CHANGELIST)
+        self.assertEqual(exp_user_data.draft_change_list_last_updated,
+                         self.NEWER_DATETIME)
+        self.assertEqual(exp_user_data.draft_change_list_exp_version, 5)
+
+    def test_create_or_update_draft_newer_draft_exists(self):
+        exp_services.create_or_update_draft(
+            self.EXP_ID1, self.USER_ID, self.NEW_CHANGELIST, 5,
+            self.OLDER_DATETIME)
+        user_exp = user_models.ExplorationUserDataModel.get(
+            self.USER_ID, self.EXP_ID1)
+        self.assertEqual(user_exp.exploration_id, self.EXP_ID1)
+        self.assertEqual(user_exp.draft_change_list, self.draft_change_list)
+        self.assertEqual(
+            user_exp.draft_change_list_last_updated, self.DATETIME)
+        self.assertEqual(user_exp.draft_change_list_exp_version, 2)
+
+    def test_create_or_update_draft_draft_does_not_exist(self):
+        with self.swap(exp_services, 'apply_change_list',
+                       self._get_exp_domain_object):
+            with self.swap(exp_domain.Exploration, 'validate',
```

Comment:
Do you need the swap here (especially if the change list you're using is valid)?